client/packages/programs/src/VaccineCourseEditModal/VaccineCourseEditModal.tsx

```diff
@@ -210,10 +215,117 @@ export const VaccineCourseEditModal: FC<VaccineCourseEditModalProps> = ({
         />
       }
       height={height}
-      width={700}
+      width={750}
       slideAnimation={false}
     >
       {modalContent}
     </Modal>
   );
 };
+
+const VaccineCourseDoseTable = ({
+  doses,
+  updatePatch,
+  courseName,
+}: {
+  courseName: string;
+  doses: VaccineCourseScheduleFragment[];
+  updatePatch: (newData: Partial<DraftVaccineCourse>) => void;
+}) => {
+  const t = useTranslation('programs');
+
+  const addDose = () => {
+    updatePatch({
+      vaccineCourseSchedules: [
+        ...doses,
+        {
+          __typename: 'VaccineCourseScheduleNode',
+          id: FnUtils.generateUUID(),
+          // temp - will be overwritten by the backend to assign unique dose number (even if previous doses were deleted)
+          doseNumber: doses.length + 1,
+          label: `${courseName} ${doses.length + 1}`,
+          minAgeMonths: 0,
+          minIntervalDays: 0,
+        },
+      ],
+    });
+  };
+
+  const deleteDose = (id: string) => {
+    updatePatch({
+      vaccineCourseSchedules: doses.filter(dose => dose.id !== id),
+    });
+  };
+
+  const updateDose: ColumnDataSetter<VaccineCourseScheduleNode> = newData => {
+    updatePatch({
+      vaccineCourseSchedules: doses.map(dose =>
+        dose.id === newData.id ? { ...dose, ...newData } : dose
+      ),
+    });
+  };
+
+  const columns = useColumns<VaccineCourseScheduleNode>(
+    [
+      {
+        key: 'doseNumber',
+        Cell: NumberCell,
+        width: 80,
+        label: 'label.dose-number',
+        accessor: ({ rowData }) => doses.indexOf(rowData) + 1,
+      },
+      {
+        key: 'label',
+        Cell: props => <TextInputCell fullWidth {...props} />,
+        width: 280,
+        label: 'label.label',
+        setter: updateDose,
+      },
+      {
+        key: 'minAgeMonths',
+        Cell: NumberInputCell,
+        label: 'label.age-months',
+        setter: updateDose,
+      },
+      {
+        key: 'minIntervalDays',
+        Cell: NumberInputCell,
+        label: 'label.min-interval',
+        setter: updateDose,
+      },
+      {
+        key: 'delete',
+        Cell: ({ rowData }) => (
+          <IconButton
+            icon={<DeleteIcon sx={{ height: '0.9em' }} />}
+            label={t('label.delete')}
+            onClick={() => deleteDose(rowData.id)}
+          />
+        ),
+      },
+    ],
+    {},
+    [updateDose, doses]
+  );
+
+  return (
+    <>
+      <Box display="flex" justifyContent="flex-end" marginBottom="8px">
+        <ButtonWithIcon
+          Icon={<PlusCircleIcon />}
+          label={t('label.dose')}
+          onClick={addDose}
+        />
+      </Box>
+      <TableProvider createStore={createTableStore}>
+        <DataTable
+          id={'doses-list'}
+          columns={columns}
+          data={doses}
+          noDataMessage={t('message.add-a-dose')}
+          dense
```

Comment:
That was easy @CarlosNZ 😁 

![Screenshot 2024-09-03 at 9 45 06 AM](https://github.com/user-attachments/assets/c4b81805-f437-4cb3-861a-75029e096546)

Reply:
Love it 👍